### PR TITLE
[Discuss] A NotificationBuilder

### PR DESCRIPTION
**Pull request by xster**
Not a real PR. Just thinking out loud. I think something like this could be useful. Worth adding to the framework?

### CODE REVIEW

**Comment by Hixie** [comment]
we should wait until we have a real need before adding it, but the idea seems sound

File: packages/flutter/lib/src/widgets/notification_listener.dart [hunk 1]
```diff
@@ -132,6 +132,34 @@ class NotificationListener<T extends Notification> extends StatelessWidget {
...
+  final NotificationChangedBuilder<T> builder;
+  final bool intercepting;
+  final Widget child;
+
+  @override
+  State<StatefulWidget> createState() => NotificationBuilderState<T>();
+}
+
+class NotificationBuilderState<T extends Notification> extends State<NotificationBuilder<T>> {
+  T lastNotification;
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/notification_listener.dart]: maybe a list? there could be many notifications.

Comment (xster) [reply packages/flutter/lib/src/widgets/notification_listener.dart]: what does it mean concretely? Does scrollables send multiple notifications per frame?

**Comment by xster** [comment]
I was hoping to use something like this in a demo for my presentation to show widget composability :)

File: packages/flutter/lib/src/widgets/notification_listener.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter/lib/src/widgets/notification_listener.dart]: They can, sure. There's nothing about the notification system that limits notifications to one per frame.

**Comment by Hixie** [comment]
What is the use case in your presentation?

**Comment by xster** [comment]
I'm trying to show that the 'middleware' in the framework are all customizable by making a ScrollView also rotate when scrolling :)

**Comment by Hixie** [comment]
@xster should we close this until we have a need for it?

**Comment by xster** [comment]
Sure

**Pull request closed without merging**